Modules/CaPTkCommon/include/CaPTkCohortOperations.h

```diff
@@ -0,0 +1,135 @@
+/** \file CaPTkCohortOperations.h
+ * \brief Defines functions that operate on cohorts
+ * 
+ * Operations either via the captk::Cohort class and similar 
+ * or via the file system with either json files or directory structures
+ */
+
+#ifndef CaPTkCohortOperations_h
+#define CaPTkCohortOperations_h
+
+// The following header file is generated by CMake and thus it's located in
+// the build directory. It provides an export macro for classes and functions
+// that you want to be part of the public interface of your module.
+#include <MitkCaPTkCommonExports.h>
+
+namespace captk
+{
+class Cohort;
+class CohortSubject;
+class CohortStudy;
+class CohortSeries;
+class CohortImage;
+}
+
+#include <QObject>
+#include <QString>
+#include <QStringList>
+#include <QList>
+#include <QSharedPointer>
+#include <QJsonDocument>
+
+#include <string>
+#include <vector>
+
+namespace captk
+{
+
+/** \brief Merge multiple cohorts into one
+ *
+ * Each subject that appears on multiple will have it's images concatenated
+ * \param cohorts list of all the cohorts to be merged
+ * \return a unified cohort
+ */
+MITKCAPTKCOMMON_EXPORT QSharedPointer<captk::Cohort> 
+CohortMerge(QList<QSharedPointer<captk::Cohort>> cohorts);
+
+/** \brief Get a QJsonDocument containing the 
+ * information of a cohort from a directory structure that
+ * follows the convention: 
+ * COHORT_ROOT/SUBJ_NAME/STUDY_NAME/MODALITY_NAME/SERIES_DESC_NAME/image.nii.gz 
+ */
+MITKCAPTKCOMMON_EXPORT QSharedPointer<QJsonDocument> 
+CohortJsonFromDirectoryStructure(QString& directory);
+
+/** \brief Load from json -> captk::Cohort */
+MITKCAPTKCOMMON_EXPORT QSharedPointer<captk::Cohort> 
+CohortJsonLoad(QSharedPointer<QJsonDocument> json);
+
+/** \brief Create a QJsonDocument from a captk::Cohort instance */
+MITKCAPTKCOMMON_EXPORT QSharedPointer<QJsonDocument>
+CohortToJson(QSharedPointer<captk::Cohort> cohort);
+
+/** \namespace internal 
+ * \brief Contains helper functions for cohort processing
+ */
+namespace internal
```

Comment:
What is the internal namespace?

Reply:
It is a relatively common "design pattern" to write functions that are not useful outside of their file.

Another way to do this is with an unnamed namespace in the implementation file, but I never liked it.